### ISSUE TITLE
macOS Catalina 

### ISSUE THREAD

**Issue by XxBlackHatxX** [etesync/etesync-dav#142]
So I'm having an issue were the application stops responding after a few minutes of inactivity it never comes back even after editing events or adding tasks to the reminders app. I have to force quit it and then launch it again.

**Comment by tasn** [comment]
Could you please run it from the command line with `-D`? That would maybe shed some light on what's going on.

Also, what do you mean by "stops responding"? Actually stops working? Because etesync-dav has built-in throttling, so if it's not syncing you should just give it a bit of time. The way to know if it's really stuck or not is accessing the web management UI. Does that work?

As for Apple reporting the app to be not responding: it's just a mac check, but it doesn't actually mean anything in this context. Though we should fix it (by following https://pyobjc.readthedocs.io/en/pyobjc-6-branch/api/module-PyObjCTools.AppHelper.html ).

**Comment by XxBlackHatxX** [comment]
To answer your questions...
1. When I type -d it says command not found.
2. "stop responding" as in when I go to activity monitor it says "Not Responding" in red. and when I add events and tasks too reminders it does not sync until I force quit and relaunch the application. I just tested the DAV Management webpage and it loads and allows me to log in. How long would I have to wait before it syncs?

Also how should I have the refresh setting set in the account section of calendars? currently it is set to every minute.

**Comment by tasn** [comment]
Hey, sorry for the slow response.

1. When I said `-D` I meant run it as `./etesync-dav -D`. Sorry for the confusion.
2. You can ignore this stop responding error as long as you can access the webpage.

Could you please try with the latest etesync-dav? I made some changes that will make sync immediate and will therefore fix this delay isssue.

**Comment by XxBlackHatxX** [comment]
Okay I had some issues update it Apple kept saying that it was corrupted no matter which version I used. I had to run this terminal command "xattr -cr"  to get past that issue but after that it ran normally. I'm not sure what that terminal command is I just googled the error I was getting and someone in a forum mentioned it.

**Comment by XxBlackHatxX** [comment]
So after trying the new version it still doesn't seem to work consistently. Would it be possible to have a manual sync button like the iOS and iPadOS apps have.  Right now I add or edit and event and one of the following will happen...
- The events reverts back to the original state deleting all my edits over the past few minutes.
- The events update on the Mac but after refreshing on iOS or iPadOS the changes are not shown on the other devices.

**Comment by tasn** [comment]
Haven't forgotten about it. I wanted to report back once I have good news to share, but we are still waiting on a release of one of our deps. See the relevant commit I made to address this: https://github.com/etesync/etesync-dav/commit/c9dab4a66ac55db12d9f8223c4963cd60ec3c6c4

What do you mean by a manual sync button? Anyhow, could you please share the output of `./etesync-dav -D` after trying to make a change from the mac? (make sure to censor it to remove private info if there's any).

**Comment by saegl5** [comment]
I am having the same issue as @XxBlackHatxX (his first comment) but am running Big Sur, EteSync DAV 0.30.4
I was thinking to add a new issue, but maybe this issue has the same root cause(s)

here is my output from `./etesync-dav -D`:

[2020-12-08 23:14:27 -0500] [94939] [INFO] Loaded default config
[2020-12-08 23:14:27 -0500] [94939] [INFO] Loaded arguments
[2020-12-08 23:14:27 -0500] [94939] [INFO] Starting Radicale
[2020-12-08 23:14:27 -0500] [94939] [INFO] auth type is 'radicale.auth.htpasswd'
[2020-12-08 23:14:27 -0500] [94939] [INFO] storage type is 'etesync_dav.radicale.storage'
[2020-12-08 23:14:27 -0500] [94939] [INFO] rights type is 'etesync_dav.radicale.rights'
[2020-12-08 23:14:27 -0500] [94939] [INFO] web type is 'etesync_dav.radicale.web'
[2020-12-08 23:14:27 -0500] [94939] [INFO] Listening on '[127.0.0.1]:37358' with SSL
[2020-12-08 23:14:27 -0500] [94939] [INFO] Listening on '[::1]:37358' with SSL
[2020-12-08 23:14:27 -0500] [94939] [INFO] Radicale server ready
Unhandled exception in thread started by <bound method Thread._bootstrap of <Thread(Thread-1, started daemon 123145354149888)>>
Traceback (most recent call last):
  File "threading.py", line 926, in _bootstrap_inner
  File "threading.py", line 870, in run
  File "etesync-dav", line 155, in machack
  File "PyObjCTools/AppHelper.py", line 261, in runConsoleEventLoop
AttributeError: 'NoneType' object has no attribute 'earlierDate_'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "tokenize.py", line 385, in find_cookie
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xcf in position 0: invalid continuation byte

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "threading.py", line 890, in _bootstrap
  File "threading.py", line 936, in _bootstrap_inner
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 167, in format_exc
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 121, in format_exception
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 508, in __init__
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 363, in extract
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/traceback.py", line 285, in line
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/linecache.py", line 16, in getline
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/linecache.py", line 47, in getlines
  File "/usr/local/Cellar/python/3.7.5/Frameworks/Python.framework/Versions/3.7/lib/python3.7/linecache.py", line 136, in updatecache
  File "tokenize.py", line 449, in open
  File "tokenize.py", line 426, in detect_encoding
  File "tokenize.py", line 390, in find_cookie
SyntaxError: invalid or missing encoding declaration for 'etesync-dav'

**Comment by saegl5** [comment]
if I edit events or add tasks to the reminders app, I don't see any error or anything unusual in the terminal

**Comment by tasn** [comment]
The dependency still haven't made a release. I just pinged the developer again: https://github.com/ronaldoussoren/pyobjc/issues/329
